File: sdks/python/relayhub/resources/api_keys.py
from __future__ import annotations

from ..http import RequestOptions, Transport
from ..types import ApiKeyCreatedResponse, ApiKeyOut


class ApiKeysResource:
    def __init__(self, transport: Transport) -> None:
        self._transport = transport

    def create(
        self,
        *,
        name: str,
        environment: str | None = None,
        scopes: list[str] | None = None,
        expires_in_days: int | None = None,
        options: RequestOptions | None = None,
    ) -> ApiKeyCreatedResponse:
        """POST /v1/api-keys -- the full `key` is only ever present on this response. Store it now; it can't be retrieved again."""
        body = {"name": name, "environment": environment, "scopes": scopes, "expires_in_days": expires_in_days}
        return self._transport.request("POST", "/v1/api-keys", body, options)

    def list(self, options: RequestOptions | None = None) -> list[ApiKeyOut]:
        """GET /v1/api-keys"""
        return self._transport.request("GET", "/v1/api-keys", None, options)

    def revoke(self, key_id: str, *, reason: str | None = None, options: RequestOptions | None = None) -> ApiKeyOut:
        """POST /v1/api-keys/{id}/revoke"""
        return self._transport.request("POST", f"/v1/api-keys/{key_id}/revoke", {"reason": reason}, options)

    def rotate(self, key_id: str, options: RequestOptions | None = None) -> ApiKeyCreatedResponse:
        """POST /v1/api-keys/{id}/rotate -- revokes the old key and issues a new one; `key` is shown once, same as create()."""
        return self._transport.request("POST", f"/v1/api-keys/{key_id}/rotate", None, options)

File: sdks/node/src/resources/api-keys.ts
import type { Transport, RequestOptions } from "../transport.js";
import type { ApiKeyCreatedResponse, ApiKeyOut } from "../types.js";

export interface CreateApiKeyParams {
  name: string;
  environment?: "test" | "live";
  scopes?: string[];
  expires_in_days?: number;
}

export class ApiKeysResource {
  constructor(private readonly transport: Transport) {}

  /** POST /v1/api-keys -- the full `key` is only ever present on this response. Store it now; it can't be retrieved again. */
  create(params: CreateApiKeyParams, options?: RequestOptions) {
    return this.transport.request<ApiKeyCreatedResponse>("POST", "/v1/api-keys", params, options);
  }

  /** GET /v1/api-keys */
  list(options?: RequestOptions) {
    return this.transport.request<ApiKeyOut[]>("GET", "/v1/api-keys", undefined, options);
  }

  /** POST /v1/api-keys/{id}/revoke */
  revoke(keyId: string, params?: { reason?: string }, options?: RequestOptions) {
    return this.transport.request<ApiKeyOut>("POST", `/v1/api-keys/${keyId}/revoke`, params, options);
  }

  /** POST /v1/api-keys/{id}/rotate -- revokes the old key and issues a new one; `key` is shown once, same as create(). */
  rotate(keyId: string, options?: RequestOptions) {
    return this.transport.request<ApiKeyCreatedResponse>("POST", `/v1/api-keys/${keyId}/rotate`, undefined, options);
  }
}
